tests/runtime/test_ipython.py

```diff
@@ -232,3 +232,87 @@ def test_ipython_package_install(temp_dir, runtime_cls, run_as_openhands):
     )
 
     _close_test_runtime(runtime)
+
+
+def test_ipython_file_editor_permissions_as_openhands(
+    temp_dir, runtime_cls, run_as_openhands
+):
+    """Test file editor permission behavior when running as different users."""
+    runtime = _load_runtime(temp_dir, runtime_cls, run_as_openhands)
+    sandbox_dir = _get_sandbox_folder(runtime)
+
+    # Create a file owned by root with restricted permissions
+    action = CmdRunAction(
+        command='sudo touch /root/test.txt && sudo chmod 600 /root/test.txt'
+    )
+    logger.info(action, extra={'msg_type': 'ACTION'})
+    obs = runtime.run_action(action)
+    logger.info(obs, extra={'msg_type': 'OBSERVATION'})
+    assert obs.exit_code == 0
+
+    # Try to view the file as openhands user - should fail with permission denied
+    test_code = "print(file_editor(command='view', path='/root/test.txt'))"
+    action = IPythonRunCellAction(code=test_code)
+    logger.info(action, extra={'msg_type': 'ACTION'})
+    obs = runtime.run_action(action)
+    logger.info(obs, extra={'msg_type': 'OBSERVATION'})
+    assert 'Permission denied' in obs.content
```

Comment:
This is curious, it failed in CI 🤔
<img width="632" alt="image" src="https://github.com/user-attachments/assets/6831be9a-ce11-452d-bf76-2f9cc58fa972">

Reply:
ahhh it fails when we are running the test *as openhands*

Reply:
Oh, hah! 😅

Reply:
works now :)